src/journal-remote/journal-upload-journal.c

```diff
@@ -249,25 +249,23 @@ static void check_update_watchdog(Uploader *u) {
         }
 }
 
-static size_t journal_input_callback(void *buf, size_t size, size_t nmemb, void *userp) {
+static int journal_input_data(void *buf, size_t size, size_t offset, void *userp) {
         Uploader *u = ASSERT_PTR(userp);
         int r;
         sd_journal *j;
         size_t filled = 0;
         ssize_t w;
 
-        assert(nmemb <= SSIZE_MAX / size);
-
         check_update_watchdog(u);
 
         j = u->journal;
 
-        while (j && filled < size * nmemb) {
+        while (j && filled < size) {
                 if (u->entry_state == ENTRY_DONE) {
                         r = sd_journal_next(j);
                         if (r < 0) {
                                 log_error_errno(r, "Failed to move to next entry in journal: %m");
-                                return CURL_READFUNC_ABORT;
+                                return r;
```

Comment:
you can shorten this to `return log_error_errno(r, …`, since log_error_errno() and related calls always return `-abs()` of the first arg you pass.